backend/middleware/audit/audit.go

```diff
@@ -99,7 +99,10 @@ func (m *mid) eventFromRequest(ctx context.Context, req interface{}, info *grpc.
 		username = claims.Subject
 	}
 
-	reqBody, err := meta.APIBody(req)
+	// Deep copy before field redaction so we do not unintentionally remove fields
+	// from the request object that were passed by reference
+	auditRequest := proto.Clone(req.(proto.Message))
```

Comment:
can we move the clone into `APIBody`?

Reply:
this would also open the door to optimize further by checking if any fields have the annotation before doing the clone.